addons/website/models/res_lang.py

```diff
@@ -15,15 +15,19 @@ def write(self, vals):
                 raise UserError(_("Cannot deactivate a language that is currently used on a website."))
         return super().write(vals)
 
-    @tools.ormcache_context(keys=("website_id",))
+    @tools.ormcache_context(keys=("website_id", "web_force_installed_langs"))
     def _get_frontend(self) -> LangDataDict:
         """ Return the available languages for current request
         :return: LangDataDict({code: LangData})
         """
         if request and getattr(request, 'is_frontend', True):
             # get languages while ignoring current language as the one in the context may be invalid
-            lang_ids = self.env['website'].get_current_website().with_context(lang=False).language_ids.sorted('name').ids
-            langs = [dict(self.env['res.lang']._get_data(id=id_)) for id_ in lang_ids]
+            if self.env.context.get('web_force_installed_langs'):
+                langs = sorted([dict(data) for data in self._get_active_by('code').values()],
+                               key=lambda lang: lang['name'])
+            else:
+                lang_ids = self.env['website'].get_current_website().with_context(lang=False).language_ids.sorted('name').ids
```

Comment:
Can't we just define `lang_ids` and leave the creation of `langs` as it was for both cases?